res/controllers/Reloop-Jockey-3-ME-scripts.js

```diff
@@ -0,0 +1,444 @@
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_clear",1);
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_clear",0);
+  } else if (value == 0x7F) {
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",1);
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",0);
+  }
+}
+
+Jockey3ME.hotcue_buttonSelect = function (control) {
+  Jockey3ME.hotc_midino = [11,12,13,14,15,16,17,18];
+  for (var i = 0; i < Jockey3ME.hotc_midino.length; i++) {
+    switch (control) {
+      case Jockey3ME.hotc_midino[i]:
+        return Jockey3ME.hotc_midino[i] - 10;
+        break;
+    }
+  };
+}
+
+Jockey3ME.hotcueClearVal_off = function() {
+  if (Jockey3ME.hotcueClearVal) {
+    Jockey3ME.hotcueClearVal = 0;
+  };
+}
+
+Jockey3ME.hotcue_clear = function (channel, control, value, status, group) {
+   if (control == 0x09 && value == 0x7F) {
+    Jockey3ME.hotcueClearVal_off();
+    Jockey3ME.hotcueClearVal = 1;
+    midi.sendShortMsg(status,control,0x01);
+   } else {
+    Jockey3ME.hotcueClearVal_off();
+    midi.sendShortMsg(status,control,0x00);
+   };
+}
+
+// Effect Sections
+Jockey3ME.effectParam = function (channel, control, value, status, group) {
+  var currentDeck = parseInt(group.substring(23,24));
+  var EncoderKnopDryWet = 0;
+  var EncoderKnopFX = 0;
+  var newVal = 0;
+  var interval = 0.04;
+  var min = 0;
+  var max = 1;
+  switch (control) {
+    case 29:
+      EncoderKnopDryWet = 1;
+      break;
+    default:
+      EncoderKnopFX = control - 29;
+      break;
+  }
+  if (!EncoderKnopDryWet) {
+    if (value == 0x41) {
+      var curVal = engine.getParameter("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "parameter" + EncoderKnopFX);
+      newVal = curVal + interval;
+      if (newVal > max) newVal = max;
+    } else {
+      var curVal = engine.getParameter("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "parameter" + EncoderKnopFX);
+      newVal = curVal - interval;
+      if (newVal < min) newVal = min;
+    }
+  } else {
+    if (value == 0x41) {
+      var curVal = engine.getParameter("[EffectRack1_EffectUnit" + currentDeck + "]", "mix");
+      newVal = curVal + interval;
+      if (newVal > max) newVal = max;
+    } else {
+      var curVal = engine.getParameter("[EffectRack1_EffectUnit" + currentDeck + "]", "mix");
+      newVal = curVal - interval;
+      if (newVal < min) newVal = min;
+    }
+  }
+  switch (engine.getValue("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "num_parameters")) {
+    case 1:
+      if (EncoderKnopFX > 1) EncoderKnopFX = 0;
+      break;
+    case 2:
+      if (EncoderKnopFX > 2) EncoderKnopFX = 0;
+      break;
+  }
+  if (EncoderKnopDryWet) {
+    engine.setParameter("[EffectRack1_EffectUnit" + currentDeck + "]", "mix", newVal);
+  } else if (EncoderKnopFX) {
+    engine.setParameter("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "parameter" + EncoderKnopFX, newVal);
+  };
```

Comment:
Like the crossfader and crossfader curve function, this function would be simpler and easier to understand if the dry/wet encoder and the parameter encoders were mapped to different functions.